dbscripts/xml/upgrade.xml

```diff
@@ -132,11 +132,11 @@
 	</upgrade>
 
 	<upgrade minversion="3.1.0.0" maxversion="3.3.9.9">
+		<migration class="PKP\migration\upgrade\v3_4_0\I7167_RemoveDuplicatedUserSettingsAndDeprecatedFields" />
```

Comment:
Moved this one to the top, the other option would be to include it on the `preflight`.

Also, about the `preflight`, I think we shouldn't be doing updates there, only checks 🤔

Reply:
> Also, about the preflight, I think we shouldn't be doing updates there, only checks

I'm OK with doing cleanup changes, but *only if* they can be run repeatedly without breaking anything.

The process of checking for bad data and the process of cleaning it up are almost the same, so the preflight migration ended up doing both. In most cases we can find and clean up at the same time; in other cases (e.g. critical issues that we can't assume we can correct automatically), we detect and fail.

Reply:
Running this before the preflight check does cause a problem, though: when the preflight fails, I get a subsequent fatal error trying to revert this migration.
```
2023-05-27 00:09:40 [A pre-flight check failed. The software was successfully upgraded to 3.3.9.9 but could not be upgraded further (to 3.4.0.0). Check and correct the error, then try again.]
2023-05-27 00:09:40 [revert migration: PKP\migration\upgrade\v3_4_0\I7167_RemoveDuplicatedUserSettingsAndDeprecatedFields]
2023-05-27 00:09:40 [downgrade for "PKP\migration\upgrade\v3_4_0\I7167_RemoveDuplicatedUserSettingsAndDeprecatedFields" unsupported: Downgrade unsupported due to removed data]
```

Reply:
This is a bug for me, after setting a save point, we shouldn't fallback the previous migrations. I've created an issue here with extra points/ideas: https://github.com/pkp/pkp-lib/issues/9030

But in this specific case, it's a cosmetic issue, that migration can be executed again.